File: frontend/src/types/attendance.ts
export interface StudentAttendance {
  id: number;
  tg_id: number;
  name: string;
  status: number; // 0 = present, 1 = Н (absent), 2 = У (excused)
  reason: string;
  is_all_day?: boolean;
}

export interface LessonDetailsResponse {
  students: StudentAttendance[];
}

export interface AttendanceUpdateRequest {
  date: string;
  time: string;
  student_id: number;
  status: number;
  reason?: string;
}

File: app/schemas/attendance.py
from pydantic import BaseModel


class AttendanceUpdateRequest(BaseModel):
    date: str
    time: str
    student_id: int
    status: int
    reason: str | None = ""


class StudentAttendanceRow(BaseModel):
    id: int
    tg_id: int
    name: str
    status: int
    reason: str
    is_all_day: bool


class LessonDetailsResponse(BaseModel):
    students: list[StudentAttendanceRow]
